### PR TITLE
Adds example of Weighted Elements complication data type.

### PR DESCRIPTION
**Pull request by garanj**
Adds `WEIGHTED_ELEMENTS` complication data type to the complication test suite.

### CODE REVIEW

File: WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/WeightedElementsDataSourceService.kt [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ *
+ * If you don't perform any suspending operations to update your complications, you can subclass
+ * [ComplicationDataSourceService] and override [onComplicationRequest] directly.
+ * (see [NoDataDataSourceService] for an example)
+ */
+@RequiresApi(Build.VERSION_CODES.TIRAMISU)
+class WeightedElementsDataSourceService : SuspendingComplicationDataSourceService() {
+
+    override suspend fun onComplicationRequest(request: ComplicationRequest): ComplicationData? {
+        if (request.complicationType != ComplicationType.WEIGHTED_ELEMENTS) {
```
Comment (yschimke) [inline WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/WeightedElementsDataSourceService.kt:52]: Is null the right result here? or NoDataComplicationData?

https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:wear/watchface/watchface-complications-data-source-ktx/src/main/java/androidx/wear/watchface/complications/datasource/SuspendingComplicationDataSourceService.kt;l=45?q=SuspendingComplicationDataSourceService

**Comment by yschimke** [comment]
Can you add a screenshot from emulator testing?

File: WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/WeightedElementsDataSourceService.kt (hunk 1 above)
Comment (garanj) [reply WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/WeightedElementsDataSourceService.kt:53]: `NoDataComplicationData` is better, you're right (confirmed with Alex)

**Comment by garanj** [comment]
Added screenshots - apologies - and fixed the return values.

**Comment by yschimke** [comment]
I wasn't certain about the null thing, just wasn't what I would have written. Hope it's right now.

**Comment by garanj** [comment]
> I wasn't certain about the null thing, just wasn't what I would have written. Hope it's right now.

Yeah, good call - I was blindly copying the pattern from previous complication data types. Alex agreed your approach was better.

**Pull request closed without merging**